src/clusterfuzz/_internal/bot/tasks/utasks/uworker_handle_errors.py

```diff
@@ -12,73 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Module for handling errors in utasks."""
-from clusterfuzz._internal.bot.tasks import setup
-from clusterfuzz._internal.bot.tasks.utasks import analyze_task
-from clusterfuzz._internal.bot.tasks.utasks import fuzz_task
-from clusterfuzz._internal.bot.tasks.utasks import minimize_task
-from clusterfuzz._internal.bot.tasks.utasks import progression_task
-from clusterfuzz._internal.bot.tasks.utasks import variant_task
+
+from typing import Callable
+from typing import Dict
+
+from clusterfuzz._internal.bot.tasks.utasks import uworker_io
 from clusterfuzz._internal.protos import uworker_msg_pb2
 
+# A handler takes an output object with its `error_type` set, and does whatever
+# it wants with it.
+ErrorHandler = Callable[[uworker_io.UworkerOutput], None]
+
+# Must use string type because of protobuf enum shenanigans.
+HandlerDict = Dict["uworker_msg_pb2.ErrorType", ErrorHandler]
+
+
+class CompositeErrorHandler(object):
```

Comment:
(object) is not necessary in Python3 iirc. I think it's implicit

Reply:
Right, python2 habits. Fixed.